libs/shared/src/main/java/io/crate/common/Objects.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.common;
+
+import javax.annotation.Nullable;
+
+public class Objects {
+
+    /**
+     * Returns the first of two given parameters that is not {@code null}, if either is,
+     * or otherwise throws a {@link NullPointerException}.
+     *
+     * @return {@code first} if it is non-null; otherwise {@code second} if it is non-null
+     * @throws NullPointerException if both {@code first} and {@code second} are null
+     */
+    public static <T> T firstNonNull(@Nullable T first, @Nullable T second) {
```

Comment:
This is `Objects.requireNonNullElse`, no?

Reply:
ah yes, indeed! will update it